be/src/Project/Web/Seller/Seller.Web/Areas/Orders/ApiControllers/OrderStatusApiController.cs

```diff
@@ -26,6 +27,17 @@ public class OrderStatusApiController : BaseApiController
             this.orderLocalizer = orderLocalizer;
         }
 
+        [HttpPost]
+        public async Task<IActionResult> Cancel([FromBody] UpdateOrderStatusRequestModel model)
+        {
+            var token = await HttpContext.GetTokenAsync(ApiExtensionsConstants.TokenName);
+            var language = CultureInfo.CurrentUICulture.Name;
+
+            var orderStatusId = await this.ordersRepository.SaveOrderStatusAsync(token, language, model.OrderId, OrdersConstants.OrderStatuses.CancelId);
```

Comment:
**suggestion:** Please add check if the order state is in new. If not, throw a custom exception with a meaningful message that only newly created orders can be cancelled.